controllers/compassmanager_controller.go

```diff
@@ -362,6 +361,45 @@ func (cm *CompassManagerReconciler) needsToBeReconciled(obj runtime.Object) bool
 	return false
 }
 
+func (cm *CompassManagerReconciler) needsToBeReconciledUpdate(oldObj, newObj runtime.Object) bool {
+	oldKymaObj, ok := oldObj.(*kyma.Kyma)
+	if !ok {
+		cm.Log.Error("Unexpected type detected. Old object is supposed to be of the Kyma type.")
+		return false
+	}
+	newKymaObj, ok := newObj.(*kyma.Kyma)
+	if !ok {
+		cm.Log.Error("Unexpected type detected. New object is supposed to be of the Kyma type.")
+		return false
+	}
+
+	oldModules := getModuleNames(oldKymaObj.Status.Modules)
+	newModules := getModuleNames(newKymaObj.Status.Modules)
+
+	if !existsInSlice(oldModules, ApplicationConnectorModuleName) && existsInSlice(newModules, ApplicationConnectorModuleName) {
+		return true
+	}
+
+	return false
+}
+
+func getModuleNames(modules []kyma.ModuleStatus) []string {
+	var result []string
```

Comment:
For performance reasons, using a map would avoid the looping over arrays - probably a bit more memory expensive as a plain array, but I asume the amount of modules will probably never grow to several thousands.

e.g.
```
cache := make (map[string]bool,len(modules))
for _, item := range modules {
    cache[item.Name] = true
}

//checking for an entry in the map
if _, exists := cache["modulename"]; !exists {  fmt.Println("This module doens't exist") }
```

Reply:
Could we postpone this implementation for a while?